### DISCUSSION TITLE
Not-an-issue: Checking the RDF from JSON-LD

### DISCUSSION THREAD

**Discussion by stuchalk** [TheBIPM/SI-Reference-Point-generation-scripts#23] [General]
## I would have put this on discussions if it were setup...

Commit  40fb338 just added Python code to create JSON_LD for units (all data for a unit and all its definitions).  This has been checked in terms of being converted to RDF and can easily be updated to work for the full list of units and an individual definition of a unit.

I thought a good idea to explain JSON-LD for those that are not familiar.  The JSON-LD spec [here](https://www.w3.org/TR/json-ld/) for reference.

A JSON-LD file is a JSON file with LD specific field names "keywords in JSON-LD" that start with the "@" symbol.  The most import keyword is "@context".  The context section is where you either define the meaning of each field name in the file, or you reference to a file that contains the the meaning definitions.  The JSON-LD file attached to this issue (added as a .txt file as .jsonld extension not allowed) has a context file [here](https://stuchalk.github.io/scidata/contexts/si.jsonld) (temporarily) and has entries for each field in the JSON-LD file assigning the semantic meaning defined in the .ttl files.  The other entries in the context section are the definition of the namespace aliases used in the document, and the base url for the SI related definitions.

To see this, open the JSON-LD file attached and copy and paste in into the JSON-LD [playground](https://json-ld.org/playground/).  This will convert the file to RDF.  You can copy and paste the RDF to a text file to see it better and I would copy it as "N-Quads".

[ampere.txt](https://github.com/TheBIPM/SI-Reference-Point-2023/files/13064164/ampere.txt)

**Comment by fmeynadier** [comment]
Hi @stuchalk , I agreee "discussions" is a useful tool, I have activated it !

**Reply by stuchalk** [reply]
Thanks for moving this topic to discussions, Frederic!

**Comment by fmeynadier** [comment]
JSON-LD playground is a neat tool, thanks for the pointer !

I checked with outputs of commands such as 

` curl http://62.161.69.201:8080/SI/units/ampere -H "Accept: application/ld+json" -o ‘ampere.jsonld'`

as you advised by email.

I have the feeling this is OK, but it is a bit hard to be sure that "everything is here". 
Concerning the content itself, having worked on xls->yaml made me aware that the type of strings with respect to being "LaTeX mode" or not may be clearer. For example, should the symbol of ampère be "A" or "$\\textrm{A}$" ? Should this be somehow encoded in the schema ?

**Comment by stuchalk** [comment]
As far as completeness I will try and check that the triples from the JSON-LD against those from the TTL.

As for the synonym we could put two fields for symbol one to be as a plain string and another as the latex representation.  Not sure how much that is needed...
